### PR TITLE
[FEATURE] Use v1 data context endpoint (cloud)

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 1]
```diff
@@ -81,18 +81,13 @@ def __init__(self):
         )
 
 
-def _extract_fluent_datasources(config_dict: dict) -> dict:
+def _transform_data_sources(config_dict: dict) -> dict:
     """
-    When pulling from cloud config, FDS and BSD are nested under the `"datasources" key`.
-    We need to extract the fluent datasources otherwise the data context will attempt eager config
-    substitutions and other inappropriate operations.
+    Transform data sources to expected shape
```
Comment (tyler-hoffman) [inline great_expectations/data_context/data_context/cloud_data_context.py]: Not that the obfuscation is new in this PR, but the name, signature, and docstring don't really tell me what the intended input/output are, and it looks like this is only called once. Maybe just inline the dict comprehension on line 267?

**Review by tyler-hoffman — COMMENTED** [review]
I'd like to see that `transform` function either updated to make things more clear, or just inline the comprehension, but otherwise, LGTM!

File: great_expectations/data_context/data_context/cloud_data_context.py (hunk 1 above)
Comment (joshua-stauffer) [reply great_expectations/data_context/data_context/cloud_data_context.py]: sure!

Comment (joshua-stauffer) [reply great_expectations/data_context/data_context/cloud_data_context.py]: [c13a3ad](https://github.com/great-expectations/great_expectations/pull/10093/commits/c13a3ad4426c1d0a332114871d6764efb2dc5cfa)

File: pyproject.toml [hunk 2]
```diff
@@ -640,6 +640,8 @@ filterwarnings = [
     # Example Actual Warning:
     # sqlalchemy.exc.RemovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
     'ignore: Deprecated API features detected! These feature\(s\) are not compatible with SQLAlchemy 2\.0\.',
+    # snowflake
+    "module: The GenericFunction 'flatten' is already registered and is going to be overridden:sqlalchemy.exc.SAWarning"
```
Comment (billdirks) [inline pyproject.toml]: Did we get to the bottom of why this is happening?

**Review by billdirks — APPROVED** [review]
LGTM!

File: pyproject.toml (hunk 2 above)
Comment (joshua-stauffer) [reply pyproject.toml]: i didn't investigate, just added this so ci would run. i believe @Kilo59 looked into it this morning and has a pr to address.

Comment (Kilo59) [reply pyproject.toml]: The warning is only emitted when the latest snowflake-sqlalchemy is installed alongside sqlalchemy 1.
There is no warning when sqlalchemy 2 is installed alongside the latest snowflake dep.


https://github.com/great-expectations/great_expectations/pull/10095#discussion_r1672196930

**Pull request merged**